tests/tools/test_openroad.py

```
@@ -1284,6 +1580,43 @@ def test_openroad_fillmetal_insertion_parameter_add_fill():
+def test_openroad_fillmetal_insertion_skips_when_disabled(asic_gcd):
+    """fin_add_fill=False must drop the node, not run a process that fills nothing.
+
+    The reason is asserted because the ordering matters: the disabled check has to
+    come before the PDK rule lookup, or a disabled task on a PDK with no rules would
+    report the wrong cause.
+    """
+    fillmetal_insertion.FillMetalTask.find_task(asic_gcd).set_openroad_addfill(False)
+
+    node = SchedulerNode(asic_gcd, "dfm.metal_fill", "0")
+    with node.runtime():
+        with pytest.raises(TaskSkip, match="^metal fill is disabled$"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)